File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Arrays.sort(nums);
        for (int i = 1; i<nums.length; i++){
            if(nums[i] != nums[i-1]){
                i++;
            }
            else{
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.py
class Solution:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        res=[]
        cur=[]
        candidates.sort()
        def dfs(start,total):
            if total==target:
                res.append(cur.copy())
                return

            for i in range(start, len(candidates)):
                if i>start and candidates[i]==candidates[i-1]:
                    continue
                if total+candidates[i]>target:
                    break
                
                cur.append(candidates[i])
                dfs(i+1,total+candidates[i])
                cur.pop()
            
        dfs(0,0)
        return res